src/main/java/org/commcare/formplayer/configuration/WebSecurityConfig.java

```diff
@@ -0,0 +1,84 @@
+package org.commcare.formplayer.configuration;
+
+import org.commcare.formplayer.auth.CommCareSessionAuthFilter;
+import org.commcare.formplayer.auth.HmacAuthFilter;
+import org.commcare.formplayer.services.FormSessionService;
+import org.commcare.formplayer.services.HqUserDetailsService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.config.annotation.authentication.builders.AuthenticationManagerBuilder;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
+import org.springframework.security.config.http.SessionCreationPolicy;
+import org.springframework.security.web.authentication.UsernamePasswordAuthenticationFilter;
+import org.springframework.security.web.authentication.preauth.PreAuthenticatedAuthenticationProvider;
+
+@Configuration
+@EnableWebSecurity
+public class WebSecurityConfig extends WebSecurityConfigurerAdapter {
+    @Autowired
+    private HqUserDetailsService userDetailsService;
+
+    @Autowired
+    private FormSessionService formSessionService;
+
+    @Value("${commcarehq.formplayerAuthKey}")
+    private String formplayerAuthKey;
+
+    @Override
+    protected void configure(HttpSecurity http) throws Exception {
+        disableDefaults(http);
+
+        http
+            .authorizeRequests()
+            .antMatchers("/serverup", "/validate_form", "/favicon.ico")
+            .permitAll();
+        http.authorizeRequests().antMatchers("/**").authenticated();
+        http.addFilterAt(getHmacAuthFilter(), UsernamePasswordAuthenticationFilter.class);
```

Comment:
I'm assuming this is why the ordering got removed above?

Reply:
not sure what ordering you mean, the comment above was about the logging filter and the sentry context filter which are both set to "1" which is immediately after the security filter chain.

The ordering of these filters is set to preserve the order of the processing in the `FormplayerAuthFilter`. I don't think the ordering matters but since the HMAC filter can tell if it should be applied by looking for the header it did make sense to put that first.

Reply:
Thanks, wasn't sure if these became the explicit ordering, or whether the system was still respecting the Annotations. Small points either way, just wanted to make sure I wasn't missing anything.

Reply:
Yea, it is a bit confusing since the security chain is a set of filters that are run 'as a unit' in the main filter chain. This diagram illustrates it well: https://docs.spring.io/spring-security/site/docs/5.4.5/reference/html5/#servlet-securityfilterchain